### PR TITLE
ci: Add Github workflow to run Ganesha and Unit Tests

### PR DESCRIPTION
**Pull request by ralcolea**
Previously, Ganesha and Unit Tests suites ran as part of a Jenkins pipeline.
This commit adds a Github Actions workflow to run Ganesha and Unit Tests suites.

### CODE REVIEW

File: .github/workflows/run-ganesha-and-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        sudo chmod o+xr /home/runner/
+        sudo chown saunafstest:saunafstest /mnt/saunafstest_loop_*
+        sudo chown saunafstest:saunafstest /mnt/hd{b,c,d}
+        chmod +x $GITHUB_WORKSPACE/tests/ci_build/run-ganesha-tests.sh
+        $GITHUB_WORKSPACE/tests/ci_build/run-ganesha-tests.sh
+
+    - name: Run Unit Tests suite
+      run: |
+        chmod +x $GITHUB_WORKSPACE/tests/ci_build/run-unit-tests.sh
+        $GITHUB_WORKSPACE/tests/ci_build/run-unit-tests.sh
```
Comment (uristdwarf) [inline .github/workflows/run-ganesha-and-unit-tests.yml:53]: Unit tests should probably be before, since they are shorter

**Review by uristdwarf — DISMISSED** [review]
LGTM

File: .github/workflows/run-ganesha-and-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+name: Run Ganesha Tests
```
Comment (uristdwarf) [inline .github/workflows/run-ganesha-and-unit-tests.yml:1]: Run Unit and Ganesha tests

Comment (ralcolea) [reply .github/workflows/run-ganesha-and-unit-tests.yml:1]: Nice catch, thank you :)

File: .github/workflows/run-ganesha-and-unit-tests.yml (hunk 1 above)
Comment (ralcolea) [reply .github/workflows/run-ganesha-and-unit-tests.yml]: I agree. Done.

**Review by uristdwarf — APPROVED** [review]

**Review by lgsilva3087 — APPROVED** [review]
LGTM

**Review by uristdwarf — APPROVED** [review]

**Review by rolysr — APPROVED** [review]
LGTM

**Pull request merged**